app/models/manageiq/providers/kubevirt/inventory/parser.rb

```diff
@@ -212,8 +214,49 @@ def process_template(object)
     template_object.vendor = ManageIQ::Providers::Kubevirt::Constants::VENDOR
 
     # Add the inventory object for the hardware:
+    process_hardware(template_object, object.spec.template.spec.domain, metadata)
+
+    # Add the inventory object for the OperatingSystem
+    process_os(template_object, metadata)
+  end
+
+  def process_hardware(template_object, domain, metadata)
     hw_object = hw_collection.find_or_build(template_object)
-    hw_object.memory_mb = ManageIQ::Providers::Kubevirt::MemoryCalculator.convert(domain.memory, 'Mi')
+    hw_object.memory_mb = ManageIQ::Providers::Kubevirt::MemoryCalculator.convert(domain.resources.requests.memory, 'Mi')
+    hw_object.cpu_cores_per_socket = domain.cpu.cores
+    hw_object.cpu_total_cores = domain.cpu.cores
+    hw_object.guest_os = metadata.labels.send("miq.github.io/kubevirt-os")
+
+    # Add the inventory objects for the disk:
+    process_disks(hw_object, domain)
+  end
+
+  def process_disks(hw_object, domain)
+    domain.devices.disks.each do |disk|
+      disk_object = disk_collection.find_or_build_by(
+        :hardware    => hw_object,
+        :device_name => disk.name
+      )
+      disk_object.device_name = disk.name
+      disk_object.location = disk.volumeName
+      disk_object.device_type = 'disk'
+      disk_object.present = true
+      disk_object.mode = 'persistent'
+      disk_object.controller_type = disk.disk.dev
+      # TODO: what do we need more? We are missing reference to PV or PVC
+    end
+  end
+
+  def process_os(template_object, metadata)
+    os_object = vm_os_collection.find_or_build(template_object)
+    os_object.product_name = metadata.labels.send("miq.github.io/kubevirt-os")
+    os_object.product_type = if metadata.annotations.tags.include?("linux")
+                               "linux"
```

Comment:
don't we have symbols for those ? or constants ?

Reply:
No we are using `open_struct`